File: experiments/nn_play.cpp
#include <iostream>
#include <GoNNPlayer.h>
#include <GoTrial.h>

int main()
{
	srand(time(NULL));

	std::string player_a_file_name = "networks/9_9_go/player_0.json";
	std::string player_b_file_name = "networks/9_9_go/player_0.json";
	unsigned int board_size = 9;

	GoNNPlayer player_a(player_a_file_name, board_size);
	GoNNPlayer player_b(player_b_file_name, board_size);

	GoTrial trial;

	trial.process(player_a, player_b, 100, board_size, 6.5, true);
	trial.print();

	std::cout << "program done\n";

 	return 0;
}

File: experiments/nn_play.py
import sys
sys.path.insert(0, "/home/michal/programming/go/libs_python")
import pygo


'''
player_a_file_name = "networks/19_19_go/player_2.json"
player_b_file_name = "networks/19_19_go/player_2.json"

board_size = 19

player_a = pygo.GoNNPlayer(player_a_file_name, board_size)
player_b = pygo.GoNNPlayer(player_b_file_name, board_size)

trial = pygo.GoTrial()

trial.process(player_a, player_b, 1000, board_size, 6.5, True)
trial._print()
'''


player_a_file_name = "networks/9_9_go/player_0.json"
player_b_file_name = "networks/9_9_go/player_1.json"

board_size = 9

player_a = pygo.GoNNPlayer(player_a_file_name, board_size)
player_b = pygo.GoNNPlayer(player_b_file_name, board_size)

trial = pygo.GoTrial()

trial.process(player_a, player_b, 1000, board_size, 6.5, True)
trial._print()
